package/base-files/files/bin/board_detect

```diff
@@ -5,8 +5,8 @@ CFG=$1
 [ -n "$CFG" ] || CFG=/etc/board.json
 
 [ -d "/etc/board.d/" -a ! -s "$CFG" ] && {
-	for a in $(ls /etc/board.d/*); do
```

Comment:
While I agree with this change, is there any reason why someone would want to do `ls <some glob>`? Probably this is done because `ls` guarantees that they are sorted which makes sense in this context.

Reply:
P.S. it does seem like `echo /etc/board.d/*` is sorted, but the `ls` man page explicitly mentions that the output would be sorted by default. I'm not sure if we can rely on it.

Reply:
The commit message for when this was introduced didn't shed any light on this.  I think @rany2 almost certainly has the right of it though.  `$(ls /etc/board.d/*)` is used since it gives the files in **sorted** order.  A favorite approach is to name the scripts `0-first-script.sh`, `1-second-script.sh`, `2-third-script.sh`, etc in order to ensure ordering of the scripts.  Same idea as patch filenames starting with 3 digits.  As a result this first line is almost certainly **wrong**.

Reply:
POSIX specifies that " The pathnames shall be in sort order as defined by the current setting of the LC_COLLATE category", and this is true for all shell implementations of globbing that I have ever seen.